ganga/GangaDirac/Lib/Utilities/DiracUtilities.py

```diff
@@ -262,6 +263,9 @@ def execute(command,
                 data = s.recv(1024)
                 out += data.decode("utf-8")
             s.close()
+            #Some regex nonsense to deal with the long representations in python 3
+            out = re.sub('(\d)L(\})', r'\1\2', out)
+            out = re.sub('(\d)L(\,)', r'\1\2', out)
```

Comment:
I don't think this is enough to catch all possible occurrences. It might be better to reduce it to one line like:
```suggestion
            out = re.sub(r'(\d)L(\,|\)|\]|\}|\:)', r'\1\2', out)
```
Or perhaps:
```suggestion
            out = re.sub(r'(\d)L([^A-Za-z0-9"\'])', r'\1\2', out)
```
The first group also risks difficult to track down bugs, imagine having an LFN called `/lhcb/my-file1L,5.root`. This should be more robust:
```suggestion
            out = re.sub(r'((?:^|\s|,|{|\()\d+)L([^A-Za-z0-9\"\'])', r'\1\2', out)
```
but it's still not perfect and things like `", 000L "` will sneak though. Finding balanced quotes is a pain and it might be better to use something like json instead of `eval` to pass between DIRAC and Ganga.

See https://regex101.com/r/fdRpED/1 for a breakdown of the last regex I suggest.

Reply:
Thanks. You are right it is pretty awful - doing it better is something we should aim for.